sounddevice.py

```diff
@@ -2657,20 +2675,26 @@ def _split(value):
     return invalue, outvalue
 
 
-def _check(err, msg=""):
+def _check(err, msg=''):
     """Raise error for non-zero error codes."""
-    if err < 0:
-        msg += ': ' if msg else ''
-        if err == _lib.paUnanticipatedHostError:
-            info = _lib.Pa_GetLastHostErrorInfo()
-            hostapi = _lib.Pa_HostApiTypeIdToHostApiIndex(info.hostApiType)
-            msg += 'Unanticipated host API {0} error {1}: {2!r}'.format(
-                hostapi, info.errorCode, _ffi.string(info.errorText).decode())
-        else:
-            msg += _ffi.string(_lib.Pa_GetErrorText(err)).decode()
-        raise PortAudioError(msg)
-    return err
+    if err >= 0:
+        return err
+
+    if msg: msg += ': '
+
+    if err == _lib.paUnanticipatedHostError:
+        info = _lib.Pa_GetLastHostErrorInfo()
+        hostapi = _lib.Pa_HostApiTypeIdToHostApiIndex(info.hostApiType)
+        apiinfo = _lib.Pa_GetHostApiInfo(hostapi)
+        apiname = _ffi.string(apiinfo.name).decode() if apiinfo else 'unknown'
+        hosterrmsg = _ffi.string(info.errorText).decode()
+        msg += "Unanticipated {0} host error {1}".format(apiname, info.errorCode)
+        if hosterrmsg:
+            msg += ": '{0}'".format(hosterrmsg)
+    else:
+        msg += _ffi.string(_lib.Pa_GetErrorText(err)).decode()
```

Comment:
I was leaning towards doing it my way because it doesn't really make sense to me to have an exception class that forces a specific error message formatting. If I want to `raise PortAudioError("This is my error message. No one elses. Mine.")` I would expect to get my error message verbatim. I don't have much experience creating Exception classes either, but I've never seen an exception that forces formatting of error messages. Hence, _check serves as the default formatter of portaudio error messages.

This is just my feeling though, ultimately I'll it up to you what you want to implement.